src/lib/DepositApiClient.js

```diff
@@ -36,10 +28,21 @@ export class DepositApiClientResponse {
 
 export class DepositApiClient {
   /* eslint-disable no-unused-vars */
-  constructor(createDraftURL, recordSerializer) {
+  constructor(apiHeaders, createDraftURL, recordSerializer) {
     if (this.constructor === DepositApiClient) {
       throw new Error("Abstract");
     }
+
+    this.apiHeaders = Object.assign({}, BASE_HEADERS, apiHeaders);
+
+    this.apiConfig = {
```

Comment:
Suggestion: make the entire config overridable, something like this (pseudocode):
```
constructor(extraApiConfig, createDraftURL, recordSerializer) {
...
}

const extraHeaders = _get(extraApiConfig, "headers", {});
this.apiHeaders = Object.assign({}, BASE_HEADERS, extraHeaders);

this.apiConfig = {
    withCredentials: true,
    ...extraApiConfig,
}
```